src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/attachment-units/attachment-units.component.ts

```diff
@@ -0,0 +1,200 @@
+    }
+
+    ngOnInit(): void {
+        this.removeBreakSlides = false;
+        this.isLoading = true;
+        this.isProcessingMode = true;
+
+        const formData: FormData = new FormData();
+        formData.append('file', this.file);
+
+        this.attachmentUnitService.getSplitUnitsData(this.lectureId, formData).subscribe({
+            next: (res: any) => {
+                if (res) {
+                    this.units = res.body.units;
+                    this.numberOfPages = res.body.numberOfPages;
+                    this.isLoading = false;
+                }
+            },
+            error: (res: HttpErrorResponse) => {
+                if (res.error.params === 'file' && res?.error?.title) {
+                    this.alertService.error(res.error.title);
+                } else {
+                    onError(this.alertService, res);
+                }
+                this.isLoading = false;
+            },
+        });
+    }
+
+    createAttachmentUnits(): void {
+        this.isLoading = true;
+        const lectureUnitInformationDTOObj: LectureUnitInformationDTO = { units: this.units, numberOfPages: this.numberOfPages, removeBreakSlides: this.removeBreakSlides };
+        const formData: FormData = new FormData();
+        formData.append('file', this.file);
+        formData.append('lectureUnitInformationDTO', objectToJsonBlob(lectureUnitInformationDTOObj));
+
+        this.attachmentUnitService.createUnits(this.lectureId, formData).subscribe({
+            next: () => {
+                this.router.navigate(['../../'], { relativeTo: this.activatedRoute });
+                this.isLoading = false;
+            },
+            error: (res: HttpErrorResponse) => {
+                if (res.error.params === 'file' && res?.error?.title) {
+                    this.alertService.error(res.error.title);
+                } else {
+                    onError(this.alertService, res);
+                }
+                this.isLoading = false;
+            },
+        });
+    }
+
+    /**
+     * Go back to the lecture page
+     */
+    cancelSplit() {
+        this.router.navigate(['course-management', this.courseId.toString(), 'lectures', this.lectureId.toString()]);
+    }
+
+    addRow() {
+        const unitDynamic = {
+            unitName: '',
+            startPage: 0,
+            endPage: 0,
+        };
+        this.units.push(unitDynamic);
+        return true;
+    }
+
+    deleteRow(i: number) {
+        if (this.units.length === 1) {
+            return false;
+        } else {
+            this.units.splice(i, 1);
+            return true;
+        }
+    }
+
+    validUnitInformation(): boolean {
+        for (const unit of this.units) {
+            if (!unit.unitName) {
+                this.invalidUnitTableMessage = this.translateService.instant(`artemisApp.attachmentUnit.createAttachmentUnits.validation.empty.unitName`);
+                return false;
+            }
+
+            if (unit.startPage === null) {
```

Comment:
This is directly connected as `ngModel` to the input field, right?
I’m not that familiar with Angular details: Is the empty case only represented as `null` or can it be `undefined` or `''`, too?

Reply:
Because `null == undefined` is `true`, the above `if (unit.startPage === null) {` will catch both null and undefined and i dont think it can be `''` as its a number.

Reply:
Be careful: `null === undefined` (triple equals) is `false`.